### DISCUSSION TITLE
Why does decoder return object instead of an array?

### DISCUSSION THREAD

**Discussion by hitheshsai** [halaxa/json-machine#51] [Q&A] [answered]
Hi,

I am trying to decode this json file but when i check its type it shows as object instead of an array.
`$results = JsonMachine::fromFile('/tmp/31.json');`

**Comment by halaxa** [comment, marked as answer]
Look at README. It is a Generator object which lazily fetches JSON items one by one when you loop the object via `foreach`.

**Reply by hitheshsai** [reply]
so my use case is to decode the Json to an array and pass it to a different function to build something like this, 
`$chartData = array(
            "cols" => array(
                array($data),
                array($settings)
            )
        );
        return json_encode($chartData);`

$data is the result array that is supposed to be decoded from json. So I am assuming json-machine wont really help my situation as I need to build the whole array from decoded object and memory crash happens again?

**Reply by halaxa** [reply]
It will help you as long as you will not put all the items in memory. If you really need all of it in memory (which you do only if you need to do random lookups), JSON Machine will not be much of a help. If you need to process it from beginning to end one by one and for example build `$chartData` just to save it to another file, then JSON Machine is the way to go.
